### PR TITLE
Add modal, button, and footer to the Code Analysis rule pages - [WEB-4510]

### PR DESCRIPTION
**Pull request by StefonSimmons**
### What does this PR do? What is the motivation?

- adds a `try this rule` cta to all static analysis rule pages. The cta should open a modal.
- adds a banner at the end of each rule page
- adds unit tests for `copy-code.js`

jira: https://datadoghq.atlassian.net/jira/software/projects/WEB/boards/102?selectedIssue=WEB-4510

preview: https://docs-staging.datadoghq.com/stefon.simmons/add-static-analysis-rule-cta/code_analysis/static_analysis_rules/
- select a rule from the page. the new cta and banner should be present.
- click on the cta. test that you are able to copy code snippets from the modal. 
- new cta and banner should **NOT** appear on `infrastructure/resource_catalog/` or `security/threats` pages pulled in from external repos
   -  https://docs-staging.datadoghq.com/stefon.simmons/add-static-analysis-rule-cta/infrastructure/resource_catalog/aws_acm/ 
   - https://docs-staging.datadoghq.com/stefon.simmons/add-static-analysis-rule-cta/security/threats/backend/

### Merge instructions

- [x] Please merge after CorpWeb and Static Analysis team reviews

### CODE REVIEW

File: content/en/code_analysis/static_analysis_rules/_index.md [hunk 1]
```diff
@@ -186,6 +186,40 @@ rulesets:
...
+    footer:
+      text: Use Datadog Code Analysis to catch code issues at every step of your development process
+      link:
+        name: Datadog Code Analysis
+        url: https://www.datadoghq.com/code-analysis/
+
+  banner: 
+    title: "<span>Seamless integrations.</span> Try Datadog Code Analysis"
+    link:
+      name: Datadog Static Analysis
```
Comment (dastrong) [inline content/en/code_analysis/static_analysis_rules/_index.md:219]:
```suggestion
      name: Datadog Code Analysis
```

I think we want this name instead

**Review by dastrong — COMMENTED** [review]
@mark-azer @juli1 
Should we Gitlab, Bitbucket, and Azure here as well?

Screenshot 2024-04-16 at 12 01 17 PM

**Comment by juli1** [comment]
> @mark-azer @juli1 Should we Gitlab, Bitbucket, and Azure here as well?
> 
> Screenshot 2024-04-16 at 12 01 17 PM

No, it's already good like this. If we put all logos, it will be too much.

**Comment by StefonSimmons** [comment]
> > @mark-azer @juli1 Should we Gitlab, Bitbucket, and Azure here as well?
> > Screenshot 2024-04-16 at 12 01 17 PM
> 
> No, it's already good like this. If we put all logos, it will be too much.

@dastrong should I still remove the `cirlceci` logo from the banner?

File: layouts/shortcodes/try-rule-banner.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{{ $ctx := . }}
+{{ $banner := $ctx.Page.Params.banner}}
+<div class="try-rule-banner">
+    <div class="try-rule-content h-100">
+        <div class="info d-flex align-items-center">
+            <div class="avatars">
+                {{ $avatars := slice "github" "circleci" "vscode" }}
```
[the remark this answers is not in the document]
Comment (StefonSimmons) [reply layouts/shortcodes/try-rule-banner.html]: @dastrong I removed the `circleci` logo from the banner

**Review by fitzage — APPROVED** [review]

**Review by dastrong — APPROVED** [review]

**Comment by dastrong** [comment]
LGTM

**Pull request merged**